addons/stock/models/stock_move_line.py

```diff
@@ -497,7 +497,8 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                 ('product_qty', '>', 0.0),
                 ('id', 'not in', ml_to_ignore.ids),
             ]
-            oudated_candidates = self.env['stock.move.line'].search(oudated_move_lines_domain)
+            current_picking_first = lambda cand: cand.picking_id != self.move_id.picking_id
```

Comment:
don't you want to do the reverse? the first move lines being in the same pickign?

Reply:
It had me confused too, `False` is 0, `True` is 1. Sorting the elements will False value firsts.